web/src/chartOptions/dataset.ts

```diff
@@ -0,0 +1,62 @@
+import {ECFeature, MaxSegmentLength, SegmentCoords, WgsCovPlotDB} from "../db";
+import {constant, isNil, sum, times, values} from "lodash";
+import {getFluGeneFeature, getMaxSegmentLength, getSegmentCoords} from "./segmented/getSegmentsInfo";
+import {setState, state} from "../state";
+import {leaveSelect} from "echarts/types/src/util/states";
+
+export const getDatasets = (db: WgsCovPlotDB) => {
+  let datasets = [];
+  if (!isNil(db.segments)) { // segmented virus
+    /* For nf-flu, setup must be done before setting chart option:
+    Set up gene feature, position array and segment coords
```

Comment:
@nhhaidee Set up these things elsewhere in a `createEffect` that is dependent on only the variables needed. Some of these things only seem to be needed to be calculated once rather than every time the datasets need to be computed.